### ISSUE TITLE
MOFA latent factors as predictor variables

### ISSUE THREAD

**Issue by zroger49** [bioFAM/MOFA2#138]
Hello! 
Thank you for development of MOFA. 

In my work, I wanted to use MOFA latent factors to predict a phenotype. 
To prevent data leakage between my train and test set, I wanted to train the MOFA model (i.e uncover features weights) using the train dataset, and then apply these weights to the test dataset.

I tried to do this with prcomp package by modifing the the weights:

```
#Remove intersect
intersect <- c(MOFA@intercepts$gexp$group1, MOFA@intercepts$meth$group1)

Y_intersect <- Y - intersect

Z <- get_factors(MOFA)
W <- get_weights(MOFA)
W <- rbind(W$gexp, W$meth)

## Use pca and stats::predict

pca <- prcomp(t(Y))
pca$rotation <- W

MOFA.predict <- stats::predict(pca, t(Y))
```

Where Y is the original data I used to train the MOFA model.

However, the factor prediction were very far from what I obtained with MOFA. 

I got similar results using matrix multiplication

```
MOFA.predict <- t(as.matrix(W)) %*% as.matrix(Y_intersect )
```

Could you help me devise a solution to my problem? I feel this might be a code issue as I am not familiar with R matrix multiplications. 

Thank in advance
Rogério Ribeiro

**Comment by 1362384737** [comment]
Hi, I met with the same problem as yours. Has it been solved yet? Thanks!

**Comment by zroger49** [comment]
No. I abandoned this idea, because even with data leakage, the models I was building with MOFA factors were not outperforming my other models
